gpiozero/output_devices.py

```diff
@@ -535,109 +536,123 @@ class TonalBuzzer(SourceMixin, CompositeDevice):
         which most users can ignore).
     """
 
-    def __init__(self, pin=None, initial_value=None, mid_note='A4', octaves=1,
-                 pin_factory=None):
-        self._validate_note_range(mid_note, octaves)
+    def __init__(self, pin=None, initial_value=None, mid_tone=Tone("A4"),
+                 octaves=1, pin_factory=None):
+        self._mid_tone = None
         super(TonalBuzzer, self).__init__(
             pwm_device=PWMOutputDevice(
                 pin=pin, pin_factory=pin_factory
             ), pin_factory=pin_factory)
         try:
+            self._mid_tone = Tone(mid_tone)
+            if not (0 < octaves <= 9):
+                raise ValueError('octaves must be between 1 and 9')
             self._octaves = octaves
+            try:
+                self.min_tone.note
+            except ValueError:
+                raise ValueError(
+                    '%r is too low for %d octaves' %
+                    (self._mid_tone, self._octaves))
+            try:
+                self.max_tone.note
+            except ValueError:
+                raise ValueError(
+                    '%r is too high for %d octaves' %
+                    (self._max_tone, self._octaves))
             self.value = initial_value
         except:
             self.close()
             raise
 
     def __repr__(self):
         try:
-            return '<gpiozero.%s object on pin %r, is_active=%s>' % (
-                self.__class__.__name__, self.pwm_device.pin, self.is_active)
+            if self.value is None:
+                return '<gpiozero.TonalBuzzer on pin %r, silent>' % (
+                    self.pwm_device.pin,)
+            else:
+                return '<gpiozero.TonalBuzzer on pin %r, playing %s>' % (
```

Comment:
Looks like you need to add ` object` into these `repr` strings to get the Travis tests passing @bennuttall

Reply:
:+1:

Reply:
Yup, done :)